aptos-move/framework/move-stdlib/sources/vector.move

```diff
@@ -263,6 +266,51 @@ module std::vector {
         }
     }
 
+    /// Apply the function to each pair of elements in the two given vectors, consuming them.
+    public inline fun zip<Element1, Element2>(v1: vector<Element1>, v2: vector<Element2>, f: |Element1, Element2|) {
```

Comment:
can you also add a version that instead returns `zip<Element1, Element2, A>(v1: vector<Element1>, v2: vector<Element2>, f: |Element1, Element2| A)`

Reply:
Added zip_map and zip_map_ref